javaScript/signup.js

```diff
@@ -0,0 +1,134 @@
+    input.style.border = "2px solid red";
+    errorElement.textContent = message;
+    errorElement.style.color = "red";
+    errorElement.style.display = "block";
+    updateSignupButtonState();
+  }
+
+  function hideError(input, errorElement) {
+    input.style.border = "";
+    errorElement.textContent = "";
+    errorElement.style.display = "none";
+    updateSignupButtonState();
+  }
+
+  function validateInput() {
+    let isValid = true;
+
+    if (emailInput.value.trim() === "" || emailError.textContent !== "") isValid = false;
+    if (usernameInput.value.trim() === "" || usernameError.textContent !== "") isValid = false;
+    if (passwordInput.value.trim() === "" || passwordError.textContent !== "") isValid = false;
+    if (confirmPasswordInput.value.trim() === "" || confirmPasswordError.textContent !== "") isValid = false;
+
+    return isValid;
+  }
+
+  function updateSignupButtonState() {
+    signupButton.disabled = !validateInput();
+  }
+
+  emailInput.addEventListener("blur", function () {
+    const emailValue = emailInput.value.trim();
+    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    if (emailValue === "") {
+      showError(emailInput, emailError, "이메일을 입력해주세요.");
+    } else if (!emailPattern.test(emailValue)) {
+      showError(emailInput, emailError, "잘못된 이메일 형식입니다.");
+    } else {
+      hideError(emailInput, emailError);
+    }
+  });
+
+  usernameInput.addEventListener("blur", function () {
+    if (usernameInput.value.trim() === "") {
+      showError(usernameInput, usernameError, "닉네임을 입력해주세요.");
+    } else {
+      hideError(usernameInput, usernameError);
+    }
+  });
+
+  passwordInput.addEventListener("blur", function () {
+    const passwordValue = passwordInput.value.trim();
+
+    if (passwordValue === "") {
+      showError(passwordInput, passwordError, "비밀번호를 입력해주세요.");
+    } else if (passwordValue.length < 8) {
+      showError(passwordInput, passwordError, "비밀번호를 8자 이상 입력해주세요.");
+    } else {
+      hideError(passwordInput, passwordError);
+    }
+  });
+
+  passwordInput.addEventListener("input", function () {
+    validateConfirmPassword();
+  });
+
+  confirmPasswordInput.addEventListener("blur", function () {
+    validateConfirmPassword();
+  });
+
+  function validateConfirmPassword() {
+    const passwordValue = passwordInput.value.trim();
+    const confirmPasswordValue = confirmPasswordInput.value.trim();
+
+    if (confirmPasswordValue === "") {
+      showError(confirmPasswordInput, confirmPasswordError, "비밀번호를 다시 입력해주세요.");
+    } else if (passwordValue !== confirmPasswordValue) {
+      showError(confirmPasswordInput, confirmPasswordError, "비밀번호가 일치하지 않습니다.");
+    } else {
+      hideError(confirmPasswordInput, confirmPasswordError);
+    }
+  }
+
+  signupButton.addEventListener("click", function (event) {
+    event.preventDefault();
+    if (!signupButton.disabled) {
+      location.href = "login.html";
+    }
+  });
```

Comment:
❗️ 수정요청
버튼이 disabled 상태이면 click 되지 않으므로 아래처럼 작성하셔도 충분할 것 같습니다.

```suggestion
  signupButton.addEventListener("click", function (event) {
    event.preventDefault();
    
    location.href = "login.html";
  });
```